### PR TITLE
Make `String` conform to `ExpressibleAs` protocols

### PR DESCRIPTION
**Pull request by kimdv**
This is some ideas I got, while I was playing around with #331 . 
Not sure if it's correct to make some if this assumptions I make in the StringConvenienceInitializers.swift?

I think we can make those assumptions because it is simple types.

But let me hear if there is anything! 
The test cases are getting more and more clean and easy to read 🎉

### CODE REVIEW

File: Sources/SwiftSyntaxBuilder/gyb_generated/Buildables.swift [hunk 1]
```diff
@@ -9548,7 +9548,7 @@ public struct SimpleTypeIdentifier: TypeBuildable {
   }
 }
 
-public protocol ExpressibleAsSimpleTypeIdentifier {
+public protocol ExpressibleAsSimpleTypeIdentifier: ExpressibleAsTypeAnnotation, ExpressibleAsTypeBuildable, ExpressibleAsTypeExpr {
```
Comment (ahoppen) [inline Sources/SwiftSyntaxBuilder/gyb_generated/Buildables.swift:9551]: Are these conformances related to making string conform to the `ExpressibleAs*` protocols?

**Review by ahoppen — COMMENTED** [review]
Nice. This is great. From a quick glance it looks to me like there are a few unrelated changes in here that were caused by a new `protocolMap.py` in the main swift repo.

File: Sources/SwiftSyntaxBuilder/gyb_generated/Buildables.swift (hunk 1 above)
Comment (kimdv) [reply Sources/SwiftSyntaxBuilder/gyb_generated/Buildables.swift:9551]: Those conformances is so we can make this possible
https://github.com/apple/swift-syntax/blob/8a51c06e76ef8311b87ef51df6742c10e38bd1ac/Tests/SwiftSyntaxBuilderTest/StructTests.swift#L53

Comment (kimdv) [reply Sources/SwiftSyntaxBuilder/gyb_generated/Buildables.swift:9551]: And there is a PR for the new `protocols.py` https://github.com/apple/swift/pull/40128

Comment (ahoppen) [reply Sources/SwiftSyntaxBuilder/gyb_generated/Buildables.swift:9551]: Ah, I see. I thought this PR was just about adding the `String: ExpressibleAsSimpleTypeIdentifier` conformances, but you went one step further to actually make this usable in places that require e.g. a `TypeAnnotation`. Looks good.

**Comment by ahoppen** [comment]
https://github.com/apple/swift/pull/40128

@swift-ci Please test

**Comment by kimdv** [comment]
Will update code when #337 is merged

**Comment by kimdv** [comment]
I have rebased this and https://github.com/apple/swift/pull/40128
Would you mind to trigger CI 🚀

**Comment by ahoppen** [comment]
@kimdv If you want, I would encourage you to request commit access to Swift (https://www.swift.org/contributing/#commit-access). I think you have accumulated more than the encouraged 5 non-trivial pull request by now. 😉 With commit access you’ll be able to trigger CI yourself.

**Comment by kimdv** [comment]
Will rebase when #334 is merged.

**Review by ahoppen — APPROVED** [review]

**Pull request merged**